src/coreclr/vm/comthreadpool.cpp

```diff
@@ -625,6 +625,19 @@ extern "C" BOOL QCALLTYPE ThreadPool_PerformGateActivities(INT32 cpuUtilization)
     return needGateThread;
 }
 
+extern "C" void QCALLTYPE ThreadPool_ThrowApplicationException(INT32 hr)
+{
+    QCALL_CONTRACT;
+
+    BEGIN_QCALL;
+
+    // This has special code to produce an error message for an HR that differs a bit from Marshal.GetExceptionForHR(). This is
```

Comment:
Is it important to maintain message level compat here, or could the tests be updated to check the new message? Assuming its throwing the same exception type.

Reply:
As we discussed offline, I'll go ahead and do the simple thing to use Marshal and remove the special casing for CoreCLR.